### PR TITLE
DG Appendix: Manual Testing

### PR DESCRIPTION
**Pull request by GnohChengYi**
This is branched from https://github.com/AY2122S1-CS2103T-W08-3/tp/pull/282, so there are a lot of overlapped changes.
So merging #282 before this will make this easier.

All new changes in this PR are in the `DeveloperGuide.md`.

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md]:
```suggestion
![Interactions within the UI Component for the `totalorders` Command](images/TotalOrdersSequenceDiagram2.png)
```

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md]:
```suggestion
![Interactions within the Logic Component for the `totalorders` Command](images/TotalOrdersSequenceDiagram1.png)
```

**Review by ngchisern — CHANGES_REQUESTED** [review]
Looks good, but is it better to add more cases for manual testing?

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md]: Thanks for pointing out the incorrect component, but I think using _Inside_ is fine.

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md]: Thanks for pointing out the incorrect component, but I think using _Inside_ is fine.

**Comment by ngchisern** [comment]
I suggest you add 2-3 more cases for manual testing or provide a more general guide for the testing. Thanks.

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md:589]: The prerequisite can be more explicit, e.g., '.... contains an empty `data` folder.'

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md:596]: The JSON file for Addressbook should be `addressBook.json` instead of `addressbook.json`.

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md:599]: This is not true because there is a possibility that the orders/tasks are wiped too if the customer of an order does not exist in the new 'empty' client list. 
If I remember correctly, there are two possibilities for this case (all data is wiped (if any order exists) or only previous tasks persist (if all the tasks are tagged with `General`))

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md:598]: `addressBook.json`

**Review by ngchisern — CHANGES_REQUESTED** [review]
Requested some changes below.

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md:589]: This test case deals with missing data. Regardless whether the `data` folder is empty, we can delete the `data` folder to simulate missing data (refer to the next line). Hence, this description is intentional, since it does not matter whether the `data` folder is empty.

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md:596]: Thanks for pointing out!

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md:598]: Thanks for pointing out!

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md:599]: Hmm seems like the behavior changed. I will update it. Thanks for pointing out.

Comment (GnohChengYi) [reply docs/DeveloperGuide.md:599]: I tried, even if all tasks are tagged with `General`, they will also be wiped out.

Comment (ngchisern) [reply docs/DeveloperGuide.md:599]: Correction:
If I remember correctly, there are two possibilities for this case (all data is wiped (if any order exists) or only previous tasks persist (if all the tasks are tagged with General and the order list is empty))

Comment (ngchisern) [reply docs/DeveloperGuide.md:599]: This is actually an implementation bug. It should behave like what I stated.

Comment (ngchisern) [reply docs/DeveloperGuide.md]: You may want to wait until #294 to be merged so that you can observe the exact behaviour.

**Comment by GnohChengYi** [comment]
Note that corrupting `salesBook.json` will start with empty data. It will wipe the data in `addressBook.json`. This is the behaviour, although unexpected. 


Output in command prompt:

image

(continued)

image

Note the last `WARNING`.

File: docs/DeveloperGuide.md
Comment (GnohChengYi) [reply docs/DeveloperGuide.md]: Okay, I will update it now.

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md:804]: In case you hadn't noticed, this is not the only expected result too.

File: docs/DeveloperGuide.md
Comment (ngchisern) [inline docs/DeveloperGuide.md:799]: Perhaps you can divide it into 2 scenarios here too.

Comment (GnohChengYi) [reply docs/DeveloperGuide.md]: Done.

**Review by ngchisern — APPROVED** [review]
LGTM!

**Pull request merged**
